expression/builtin_json.go

```diff
@@ -685,8 +689,101 @@ type jsonValidFunctionClass struct {
 	baseFunctionClass
 }
 
-func (c *jsonValidFunctionClass) getFunction(ctx sessionctx.Context, args []Expression) (builtinFunc, error) {
-	return nil, errFunctionNotExists.GenWithStackByArgs("FUNCTION", "JSON_VALID")
+func (c *jsonValidFunctionClass) getFunction(ctx sessionctx.Context, args []Expression) (sig builtinFunc, err error) {
+	if err := c.verifyArgs(args); err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	var bf baseBuiltinFunc
+	switch args[0].GetType().Tp {
+	case mysql.TypeNull:
+		bf = newBaseBuiltinFuncWithTp(ctx, args, types.ETInt, args[0].GetType().EvalType())
+		sig = &builtinJSONValidNullSig{bf}
+		sig.setPbCode(tipb.ScalarFuncSig_JsonValidSig)
+	case mysql.TypeJSON:
+		bf = newBaseBuiltinFuncWithTp(ctx, args, types.ETInt, types.ETJson)
+		sig = &builtinJSONValidJSONSig{bf}
+		sig.setPbCode(tipb.ScalarFuncSig_JsonValidSig)
+	case mysql.TypeString, mysql.TypeVarString, mysql.TypeVarchar, mysql.TypeBlob,
+		mysql.TypeLongBlob, mysql.TypeMediumBlob, mysql.TypeTinyBlob:
+		bf = newBaseBuiltinFuncWithTp(ctx, args, types.ETInt, types.ETString)
+		sig = &builtinJSONValidStringSig{bf}
+		sig.setPbCode(tipb.ScalarFuncSig_JsonValidSig)
+	default:
+		bf = newBaseBuiltinFuncWithTp(ctx, args, types.ETInt, args[0].GetType().EvalType())
+		sig = &builtinJSONValidOthersSig{bf}
+		sig.setPbCode(tipb.ScalarFuncSig_JsonValidSig)
```

Comment:
we must set different PbCode for different sigs.

Reply:
Hi, please take a look at [tipb#106](https://github.com/pingcap/tipb/pull/106)

Reply:
@kuafou Sorry for the late reply, https://github.com/pingcap/tipb/pull/106 is merged, could you update to the latest tipb and update this PR?

Reply:
I am working on it.